source/MissionAction.cpp

```diff
@@ -186,11 +196,13 @@ void MissionAction::Load(const DataNode &node, const string &missionName)
 			conversation.Load(child);
 		else if(key == "conversation" && hasValue)
 			stockConversation = GameData::Conversations().Get(child.Token(1));
+		else if(key ==  "ship" && hasValue)
```

Comment:
`ship "some model"` is not a sufficient interface for this functionality.

Unlike outfits, there is not a simple way to reflect the direction of the transfer (e.g. by passing a negative count).

Please refactor this to account for  all of the desired ship transference mechanisms (e.g. give a ship, take a ship, has a ship).

I'm certain several content creators can work with you to determine a better, more flexible syntax.

Reply:
`ship <action> "<model>" [<name>]`?

Reply:
That sounds like a resonable way to put it.

Reply:
A drawback to that syntax is the inability to change the implementation to store instantiated ship objects without breaking compatibility.

I think using the action first allows us more flexibility for future improvements.

Reply:
What?

Reply:
So, something like this? (Based on Tehhowch's suggestion on discord
```
(give|take|require)
  ship <model name> [<name>] [<InvisibleIdentifier>] [<LockShip>] [<LockOutfits>]
  outfit <outfit name> [#count]
(give|take|require) [....inline specifier]
```
InvisibleIdentifier = identifier that is stored invisibly in the ship's attributes so that the player can change the ship's name and still be identified by the mission
LockShip = variable the specifies that the ship's name is locked. (recorded in the ship's attributes)
LockOutfit = variable that specifies that the ship's outfits cannot be modified (recorded in the ship's attributes)